pkg/alignchecker/alignchecker.go

```diff
@@ -46,67 +39,84 @@ func CheckStructAlignments(pathToObj string, toCheck map[string][]reflect.Type,
 	return nil
 }
 
-// structInfo contains C struct info
 type structInfo struct {
-	size         int64
-	fieldOffsets map[string]int64
+	size         uint32
+	fieldOffsets map[string]uint32
 }
 
-func getStructInfosFromDWARF(d *dwarf.Data, toCheck map[string][]reflect.Type) (map[string]structInfo, error) {
-	structs := make(map[string]structInfo)
-
-	r := d.Reader()
+func getStructInfosFromBTF(types *btf.Spec, toCheck map[string][]reflect.Type) (map[string]*structInfo, error) {
+	structs := make(map[string]*structInfo)
+	for name := range toCheck {
+		t, err := types.AnyTypeByName(name)
+		if err != nil {
+			return nil, fmt.Errorf("looking up type %s by name: %w", name, err)
+		}
 
-	for entry, err := r.Next(); entry != nil && err == nil; entry, err = r.Next() {
-		t, err := d.Type(entry.Offset)
+		si, err := getStructInfoFromBTF(t)
```

Comment:
It looks like this changes the high level logic:

Previously we check all symbols in the binary, then look into `toCheck` to map those to the Go types and validate the alignment.

Now it looks like we only look through `toCheck`. This means that if someone forgets to update `toCheck` when introducing a new type, we would not validate the alignment between the Go / C structures.

Am I correct in the above reading?

Reply:
I think not, previously and now we  check symbols  manually added in `bpf/bpf_alignchecker.c` and `pkg/datapath/alignchecker/alignchecker.go`, not all symbols in datapath binary.

Reply:
>Previously we check all symbols in the binary, then look into toCheck to map those to the Go types and validate the alignment.

This might've been the case for map key and value structs. 

>we should switch this back so that we make sure that all structures which are shared between the datapath and Go code are checked automatically.

:heavy_plus_sign: It might be useful to have some kind of annotations to indicate that users want alignment check enforcements for certain fields. Currently, there is no central place where users can add new types, and it's quite confusing.